### PR TITLE
check-zfs: init at 2.0

### PR DESCRIPTION
**Pull request by mariaa144**
###### Description of changes

Adding check-zfs package which is a Nagios plugin for check zfs pool health.

Also added myself as a maintainer.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/monitoring/plugins/zfs.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3
+, zfs
+, sudo
+}:
+
+let
+  n="check_zfs";
```
Comment (azahi) [inline pkgs/servers/monitoring/plugins/zfs.nix:10]: Please reduce the scope of this variable to just `installPhase`.

File: pkgs/servers/monitoring/plugins/zfs.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  version = "2.0";
+
+  src = fetchFromGitHub {
+    owner  = "zlacelle";
+    repo   = "nagios_check_zfs_linux";
+    rev    = "33bea0671953a8891b46db6951f56c59a120a16c";
+    sha256 = "gPLCNt6hp4E94s9/PRgsnBN5XXQQ+s2MGcgRFeknXg4=";
+  };
+
+  buildInputs = [python3 zfs sudo];
```
Comment (azahi) [inline pkgs/servers/monitoring/plugins/zfs.nix:22]:
```suggestion
  buildInputs = [ python3 zfs sudo ];
```

File: pkgs/servers/monitoring/plugins/zfs.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  n="check_zfs";
+in stdenv.mkDerivation {
+  pname = "check-zfs";
+  version = "2.0";
+
+  src = fetchFromGitHub {
+    owner  = "zlacelle";
+    repo   = "nagios_check_zfs_linux";
+    rev    = "33bea0671953a8891b46db6951f56c59a120a16c";
+    sha256 = "gPLCNt6hp4E94s9/PRgsnBN5XXQQ+s2MGcgRFeknXg4=";
```
Comment (azahi) [inline pkgs/servers/monitoring/plugins/zfs.nix:19]:
```suggestion
    owner = "zlacelle";
    repo = "nagios_check_zfs_linux";
    rev = version;
    sha256 = "gPLCNt6hp4E94s9/PRgsnBN5XXQQ+s2MGcgRFeknXg4=";
```

You can use `rec` to make `version` here inheritable.

File: pkgs/servers/monitoring/plugins/zfs.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    sed -i 's#/usr/bin/python3#${python3}/bin/python#g' check_zfs.py
+    sed -i 's#/usr/bin/sudo#${sudo}/bin/sudo#g' check_zfs.py
+    sed -i 's#/sbin/zpool#${zfs}/bin/zpool#g' check_zfs.py
+    sed -i 's#/sbin/zfs#${zfs}/bin/zfs#g' check_zfs.py
+    cp ${n}.py $out/bin/${n}
+    chmod +x $out/bin/${n}
+  '';
```
Comment (azahi) [inline pkgs/servers/monitoring/plugins/zfs.nix:32]: Please add pre/post hooks.

Please use `install(1)` instead of `mkdir`/`cp` to install files into the derivation output. You can pass `-m 755` to make a file executable.

Please use `substituteInPlace` instead of `sed`.

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -7857,6 +7857,12 @@
     githubId = 26470037;
     name = "Markus Kowalewski";
   };
+  mariaa144 = {
```
Comment (wozeparrot) [inline maintainers/maintainer-list.nix:7860]: This should be in a separate commit.

File: pkgs/servers/monitoring/plugins/zfs.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+  meta = with lib; {
+    description = "Check the health, capacity, fragmentation, and other things for use with Nagios monitoring";
+    homepage = "https://github.com/zlacelle/nagios_check_zfs_linux";
+    license = licenses.gpl3;
```
Comment (azahi) [inline pkgs/servers/monitoring/plugins/zfs.nix:37]: This should be either `gpl3Only` or `gpl3Plus`.

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -35669,6 +35669,8 @@ with pkgs;
 
   check-uptime = callPackage ../servers/monitoring/plugins/uptime.nix { };
 
+  check-zfs = callPackage ../servers/monitoring/plugins/zfs.nix { };
```
Comment (azahi) [inline pkgs/top-level/all-packages.nix:35672]: If this is a Nagios plugin, it should be added to `servers/monitoring/nagios/plugins`.

File: pkgs/servers/monitoring/plugins/zfs.nix [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+, fetchFromGitHub
+, python3
+, zfs
+, sudo
+}:
+
+let
+  n="check_zfs";
+in stdenv.mkDerivation {
+  pname = "check-zfs";
```
Comment (azahi) [inline pkgs/servers/monitoring/plugins/zfs.nix:12]: Just a suggestion: I think it's better to leave it as `check_zfs` and not rename it.

**Review by azahi — CHANGES_REQUESTED** [review]

**Comment by mariaa144** [comment]
Thank you for the review! I tried to apply all the changes suggested. My maintainer PR is here: #181637

File: pkgs/servers/monitoring/nagios/plugins/zfs.nix [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+     substituteInPlace ${n}.py \
+                       --replace /usr/bin/python3 ${python3}/bin/python \
+                       --replace /usr/bin/sudo ${sudo}/bin/sudo \
+                       --replace /sbin/zpool ${zfs}/bin/zpool \
+                       --replace /sbin/zfs ${zfs}/bin/zfs
+     install -D -m 755 ${n}.py $out/bin/${n}
+
+     runHook preInstall
+    '';
```
Comment (azahi) [inline pkgs/servers/monitoring/nagios/plugins/zfs.nix:35]:
```suggestion
  installPhase = ''
    runHook preInstall

    substituteInPlace ${pname}.py \
      --replace /usr/bin/python3 ${python3}/bin/python \
      --replace /usr/bin/sudo ${sudo}/bin/sudo \
      --replace /sbin/zpool ${zfs}/bin/zpool \
      --replace /sbin/zfs ${zfs}/bin/zfs
    install -Dm755 ${pname}.py $out/bin/${pname}

    runHook postInstall
  '';
```

**Review by azahi — CHANGES_REQUESTED** [review]
You should leave the commit that adds yourself as a maintainer here. There's no need to create a separate PR. Make sure you have two separate commits with correct commit messages, though.

File: pkgs/servers/monitoring/nagios/plugins/zfs.nix (hunk 9 above)
Comment (mariaa144) [reply pkgs/servers/monitoring/nagios/plugins/zfs.nix:35]: Looks good!

**Comment by mariaa144** [comment]
I noticed my package in `all-packages.nix` is in the wrong place. Let me fix that.

File: pkgs/servers/monitoring/nagios/plugins/zfs.nix [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    sha256 = "gPLCNt6hp4E94s9/PRgsnBN5XXQQ+s2MGcgRFeknXg4=";
+  };
+
+  buildInputs = [ python3 zfs sudo ];
+
+  installPhase = ''
+    runHook preInstall
+
+    substituteInPlace ${pname}.py \
+      --replace /usr/bin/python3 ${python3}/bin/python \
```
Comment (AndersonTorres) [inline pkgs/servers/monitoring/nagios/plugins/zfs.nix:26]: Remove this line and add something like

```nix
postPatch = ''
  patchShebangs check_zfs.py
'';
```

File: pkgs/servers/monitoring/nagios/plugins/zfs.nix [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    rev = version;
+    sha256 = "gPLCNt6hp4E94s9/PRgsnBN5XXQQ+s2MGcgRFeknXg4=";
+  };
+
+  buildInputs = [ python3 zfs sudo ];
+
+  installPhase = ''
+    runHook preInstall
+
+    substituteInPlace ${pname}.py \
```
Comment (AndersonTorres) [inline pkgs/servers/monitoring/nagios/plugins/zfs.nix:25]: Remove this reference to `pname`. Use `check_zfs` directly.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/servers/monitoring/nagios/plugins/zfs.nix (hunk 11 above)
Comment (mariaa144) [reply pkgs/servers/monitoring/nagios/plugins/zfs.nix:25]: I can do that but what is the reason?

File: maintainers/maintainer-list.nix (hunk 5 above)
Comment (mariaa144) [reply maintainers/maintainer-list.nix:7866]: I put this in a separate commit.

File: pkgs/servers/monitoring/nagios/plugins/zfs.nix (hunk 11 above)
Comment (AndersonTorres) [reply pkgs/servers/monitoring/nagios/plugins/zfs.nix:25]: First, it is ugly. Not a good technical reason, I know, but aesthetics is important.

Second, it is not always that the installed program name has the same name as the project name.
The most common example I can provide is a fork.
Indeed some people fork abandoned projects, changing their `pname` but keeping the file tree mostly intact. If by any reason we Nix packagers decide to change for such a "revamped fork", such ${pname} parameterization will be patched.

Third, usually we avoid unneeded and/or excessive parameterization (does this word exist?). 
`version` is parameterized because it usually composes useful and changing resources as links, tags, changelogs and alike, whereas `pname` does not have the same relationship with the code.

**Review by AndersonTorres — APPROVED** [review]
Welcome aboard!
